src/dataServices/hybridDataService.ts

```diff
@@ -0,0 +1,290 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable class-methods-use-this */
+
+import uuidv4 from 'uuid/v4';
+import {
+    BulkDataAccess,
+    GenericResponse,
+    Persistence,
+    clone,
+    ReadResourceRequest,
+    vReadResourceRequest,
+    CreateResourceRequest,
+    DeleteResourceRequest,
+    UpdateResourceRequest,
+    PatchResourceRequest,
+    ConditionalDeleteResourceRequest,
+    InitiateExportRequest,
+    GetExportStatusResponse,
+    ResourceNotFoundError,
+} from 'fhir-works-on-aws-interface';
+
+import { Buffer } from 'buffer';
+import assert from 'assert';
+import { DynamoDbDataService } from './dynamoDbDataService';
+import { DynamoDbUtil } from './dynamoDbUtil';
+
+import S3ObjectStorageService from '../objectStorageService/s3ObjectStorageService';
+import { SEPARATOR } from '../constants';
+
+export const decode = (str: string): string => Buffer.from(str, 'base64').toString('utf-8');
+export const encode = (str: string): string => Buffer.from(str, 'utf-8').toString('base64');
+
+export type StripPayloadFunction = {
+    (resource: any): any;
+};
+
+export class HybridDataService implements Persistence, BulkDataAccess {
+    updateCreateSupported: boolean = false;
+
+    private resourceTypesToStoreOnObjectStorage: Map<string, StripPayloadFunction> = new Map<
+        string,
+        StripPayloadFunction
+    >();
+
+    readonly enableMultiTenancy: boolean;
+
+    private readonly dbPersistenceService: DynamoDbDataService;
+
+    private static async replaceStrippedResourceWithS3Version(strippedResource: any): Promise<any | undefined> {
```

Comment:
From your code, it looks like it would be better to always return an object. If there is nothing at s3, just return the `strippedResource`